awx/main/utils/common.py

```diff
@@ -1141,12 +1141,16 @@ def create_partition(tblname, start=None, end=None, partition_label=None, minute
         else:
             partition_label = start.strftime('%Y%m%d_%H')
 
-    with connection.cursor() as cursor:
-        cursor.execute(
-            f'CREATE TABLE IF NOT EXISTS {tblname}_{partition_label} '
-            f'PARTITION OF {tblname} '
-            f'FOR VALUES FROM (\'{start_timestamp}\') to (\'{end_timestamp}\');'
-        )
+    try:
+        with transaction.atomic():
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    f'CREATE TABLE IF NOT EXISTS {tblname}_{partition_label} '
+                    f'PARTITION OF {tblname} '
+                    f'FOR VALUES FROM (\'{start_timestamp}\') to (\'{end_timestamp}\');'
+                )
+    except ProgrammingError as e:
+        logger.debug(e)
```

Comment:
A little more context would be appreciated, because you don't get line numbers or anything. If a lot statement looks absurdly redundant, you're doing it right. Like `f'Caught known error due to existing partition: {e}'`